core/src/main/java/org/apache/calcite/avatica/DriverVersion.java

```diff
@@ -138,6 +139,14 @@ public static DriverVersion load(
       }
     } catch (IOException e) {
       e.printStackTrace();
+    } finally {
+      if (inStream != null) {
+        try {
+          inStream.close();
+        } catch (IOException ignore) {
+          ignore.printStackTrace();
```

Comment:
Could you re-throw the caught IOException as a `RuntimeException`? I can't think of a situation where this should ever fail. I'm not sure if we want this printed to stdout like you have it doing either (get a logger instance if you think this would be important. I think it would be fine to just re-throw this)

Reply:
For the test case, I mentioned the reproducing and verifying steps at https://github.com/leonliao/calcite-avatica-driver-version-test. And I didn't figure out a way to verify a native memory leak fix with just Java code. 

For the exception handling, after validating the minimum Java version is Java 8, I updated the code to use try-with-resources, then the previouslly existing IOException catch is reused.